documentation/code-generation.md

```diff
@@ -94,3 +94,80 @@ generator –r –v –l=temp –version=v11.2.0-beta C:/azure-rest-api-specs/sp
 ## Generate Azure SDK for Go profiles
 
 Take a look into the [profile generator documentation](https://github.com/Azure/azure-sdk-for-go/tree/main/eng/tools/profileBuilder)
+
+## Generate Go Mgmt SDK from Typespec
+### Prerequisites
+- [Node.js 18.x LTS](https://nodejs.org/en/download) or later
+- [Go 1.23.x](https://go.dev/doc/install) or later
+- [Git](https://git-scm.com/downloads)
+- Identify the `tspconfig.yaml` file for your package in the [Rest API Spec Repo](https://github.com/Azure/azure-rest-api-specs) and ensure there is a configuration for the Go SDK similar to that shown below and in [this example](https://github.com/Azure/azure-rest-api-specs/blob/b09c9ec927456021dc549e111fa2cac3b4b00659/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml#L40)
+
+     ```yaml
+     options:
+       "@azure-tools/typespec-go":
+         service-dir: "SERVICE_DIRECTORY_NAME"
+         package-dir: "PACKAGE_DIRECTORY_NAME"
+         module: "github.com/Azure/azure-sdk-for-go/{service-dir}/{package-dir}"
+         fix-const-stuttering: true
+         flavor: "azure"
+         generate-examples: true
+         generate-fakes: true
+         head-as-boolean: true
+         inject-spans: true
+         remove-unreferenced-types: true
+    ```
+- Local Clone of Rest API Spec Repo Fork
+  - If you don't already have a fork, [Fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo#forking-a-repository) the [Rest API Spec Repo](https://github.com/Azure/azure-rest-api-specs).
+  - Clone your fork of the repo.
+
+    ```
+      git clone https://github.com/{YOUR_GITHUB_USERNAME}/azure-rest-api-specs.git
+    ```
+- Local Clone of Go Language Repo Fork
+  - If you don't already have a fork, [Fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo#forking-a-repository) the [Go Repo](https://github.com/Azure/azure-sdk-for-go).
+  - Clone your fork of the repo.
+
+    ```
+      git clone https://github.com/{YOUR_GITHUB_USERNAME}/azure-sdk-for-go.git
+    ```
+
+### Steps
+1. Complete the prerequisites listed above
+2. Run [automation_init.sh](https://github.com/Azure/azure-sdk-for-go/blob/main/eng/scripts/automation_init.sh)
+ 
+   ```sh
+     cd "~/azure-sdk-for-go/eng/scripts" # navigate to the script directory
+     ./automation_init.sh
+   ```
+   > On windows you can use Git Bash to run the script. This script also installs the generator as a global tool.
+3. Create a local json file named `generatedInput.json` with content similar to that shown below
+
+   ```json
+      {
+        "dryRun": false,
+        "specFolder": "LOCAL_AZURE-REST-API-SPECS_REPO_ROOT", // e.g "C:\git\azure-sdk-for-go"
+        "headSha": "SHA_OF_AZURE-REST-API-SPECS_REPO", // use ' git rev-parse HEAD '
```

Comment:
How about add details in terms of where to run the 'git rev-parse HEAD' to get the `headSha`?